Common/src/main/java/mezz/jei/common/transfer/BasicRecipeTransferHandlerServer.java

```diff
@@ -38,7 +39,7 @@ public static void setItems(
 			return;
 		}
 
-		Map<Slot, ItemStack> recipeSlotToRequiredItemStack = calculateRequiredStacks(transferOperations, player);
+		Map<Slot, Pair<Slot, ItemStack>> recipeSlotToRequiredItemStack = calculateRequiredStacks(transferOperations, player);
```

Comment:
Could you make this `Pair<Slot, ItemStack>` into a Record with named fields like `hintSlot, requiredStack`, in order to clarify the code? Any code that accesses this pair's left or right gets pretty opaque.